test/protobuf/protoc/generator_integration_test.exs

```diff
@@ -48,6 +48,49 @@ defmodule Protobuf.Protoc.GeneratorIntegrationTest do
     end)
   end
 
+  test "encoding maps ensure that validates the input" do
+    cases = [
+      %{key: :int32_map, wrong_value: "not int32", expected_type: :int32},
+      %{key: :sint32_map, wrong_value: "not sint32", expected_type: :sint32},
+      %{key: :sfixed32_map, wrong_value: "not sfixed32", expected_type: :sfixed32},
+      %{key: :fixed32_map, wrong_value: "not fixed32", expected_type: :fixed32},
+      %{key: :uint32_map, wrong_value: "not uint32", expected_type: :uint32},
+      %{key: :int64_map, wrong_value: "not int64", expected_type: :int64},
+      %{key: :sint64_map, wrong_value: "not sint64", expected_type: :sint64},
+      %{key: :sfixed64_map, wrong_value: "not sfixed64", expected_type: :sfixed64},
+      %{key: :fixed64_map, wrong_value: "not fixed64", expected_type: :fixed64},
+      %{key: :uint64_map, wrong_value: "not uint64", expected_type: :uint64},
+      %{key: :float_map, wrong_value: "not float", expected_type: :float},
+      %{key: :double_map, wrong_value: "not double", expected_type: :double},
+      %{key: :string_map, wrong_value: %{levels: %{level_1: 0.4}}, expected_type: :string},
+      %{key: :bool_map, wrong_value: "not bool", expected_type: :bool},
+      %{key: :bytes_map, wrong_value: 42, expected_type: :bytes},
+      %{key: :enum_map, wrong_value: "not an map enum value", expected_type: :enum, opts: [use_enum_numbers: false]},
+
+      # ASK: I got stuck here due to getting confused by https://github.com/elixir-protobuf/protobuf/blob/885f39e2eccd5468d885d62955fcf695ac6f1fce/lib/protobuf/json/encode.ex#L213
+      %{key: :enum_map, wrong_value: :NOT_VALID, expected_type: :enum, opts: [use_enum_numbers: true]},
```

Comment:
@ericmj @whatyouhide hey peeps, I got stuck fixing the issue, I am getting the following issue

```
stacktrace:
       (protobuf 0.11.0) My.Test.MapEnum.value(:NOT_VALID)
       (protobuf 0.11.0) lib/protobuf/json/encode.ex:193: anonymous fn/5 in Protobuf.JSON.Encode.encode_value/3
       (stdlib 3.17.2) maps.erl:410: :maps.fold_1/3
       (protobuf 0.11.0) lib/protobuf/json/encode.ex:191: Protobuf.JSON.Encode.encode_value/3
       (protobuf 0.11.0) lib/protobuf/json/encode.ex:139: anonymous fn/4 in Protobuf.JSON.Encode.encode_regular_fields/3
       (stdlib 3.17.2) maps.erl:410: :maps.fold_1/3
       (protobuf 0.11.0) lib/protobuf/json/encode.ex:119: Protobuf.JSON.Encode.encode_regular_fields/3
       (protobuf 0.11.0) lib/protobuf/json/encode.ex:112: Protobuf.JSON.Encode.encodable/2
       (protobuf 0.11.0) lib/protobuf/json.ex:205: Protobuf.JSON.to_encodable/2
       (protobuf 0.11.0) lib/protobuf/json.ex:160: Protobuf.JSON.encode/2
       (protobuf 0.11.0) lib/protobuf/json.ex:110: Protobuf.JSON.encode!/2
       test/protobuf/protoc/generator_integration_test.exs:85: anonymous fn/2 in Protobuf.Protoc.GeneratorIntegrationTest."test encoding maps ensure that validates the input"/1
       (elixir 1.13.3) lib/enum.ex:2396: Enum."-reduce/3-lists^foldl/2-0-"/3
       test/protobuf/protoc/generator_integration_test.exs:82: (test)

```

I am not sure how I should be validating the enum because the comment confuses me in terms of what it meant and what to do

I would appreciate some help understanding what I should be doing for enum.